src/common/src/range.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::{Add, Bound, RangeBounds, Sub};
+
+mod private {
+
+    pub trait ZeroOne {
```

Comment:
Found a crate that implements similar functionality. 

https://docs.rs/num/latest/num/traits/trait.Zero.html

Reply:
Yep, there is `One` and `Zero`, but I need both of them. 🤣 I preferred not to introduce two more dependencies here.